docs/sphinx/applications/python/vqe_advanced.ipynb

```diff
@@ -367,7 +367,7 @@
     "                                qubit_count,\n",
     "                                electron_count,\n",
     "                                xplus[i],\n",
-    "                                qpu_id=qpu_counter))\n",
+    "                                qpu_id=qpu_counter%num_qpus))\n",
```

Comment:
I believe this change (and the one on line 379) would make `qpu_counter` an unused variable, so I'm wondering if they should be reverted.

Reply:
After looking at this again, I realize my confusion. The use of `qpu_counter` is confusing. The code was originally assuming that num_qpus was always at least 2, and that is obviously a bad assumption. Your change fixes that, which is good.

However, lines 382-383 can potentially mislead someone into thinking that qpu_counter will always reset back to 0 every time it wraps around, but that isn't actually what would happen. For example, if num_qpus was 3, the line 383 would _not_ execute when i==0 but _would_ execute when i==1. This qpu_counter variable is a bit flawed in my opinion.

In order to prevent continued confusion, I would recommend either a) duplicating lines 382-383 right after line 371, or b) removing lines 382-383. It makes no sense to have the wrap-around check after one increment but not another increment.

If you choose a), then lines 370 and 379 should simply be `qpu_id=qpu_counter` (because qpu_counter would be guaranteed to be < num_qpus), and if you choose b), then lines 370 and 379 should be as you have them right now (`qpuid=qpu_counter%num_qpus`).